crates/shared/src/ethrpc.rs

```diff
@@ -5,24 +5,88 @@ pub mod http;
 pub mod mock;
 
 use self::{buffered::BufferedTransport, http::HttpTransport};
-use crate::http_client::HttpClientFactory;
+use crate::{arguments::duration_from_seconds, http_client::HttpClientFactory};
 use ethcontract::{batch::CallBatch, dyns::DynWeb3, transport::DynTransport};
 use reqwest::{Client, Url};
-use std::convert::TryInto as _;
+use std::{
+    fmt::{self, Display, Formatter},
+    num::NonZeroUsize,
+    time::Duration,
+};
 
 pub const MAX_BATCH_SIZE: usize = 100;
 
 pub type Web3 = DynWeb3;
 pub type Web3Transport = DynTransport;
 pub type Web3CallBatch = CallBatch<Web3Transport>;
 
+/// Command line arguments for the common Ethereum RPC connections.
+#[derive(clap::Parser)]
+#[group(skip)]
+pub struct Arguments {
+    /// Maximum batch size for Ethereum RPC requests. Use '0' to disable batching.
+    #[clap(long, default_value = "100")]
+    pub ethrpc_max_batch_size: usize,
+
+    /// Maximum number of concurrent requests to send to the node. Use '0' for
+    /// no limit on concurrency.
+    #[clap(long, default_value = "1")]
+    pub ethrpc_max_concurrent_requests: usize,
```

Comment:
Do we want this default to 1? This is kind of a rate limit where we before we didn't have a rate limit.

Reply:
We can default to 0 (so no limits) if you think that is more appropriate.

Reply:
I'm fine either way just wanted to note that if you merge and we don't set it to 0 in kubernetes we might have slow requests.

Reply:
Changed it to 10 concurrent batches. This is what we've been testing with in our load-balancing setup as our "under heavy load" setup.